libretroshare/src/pqi/pqifdbin.cc

```diff
@@ -53,8 +54,11 @@ void RsFdBinInterface::setSocket(int s)
 
 #else
     // On windows, there is no way to determine whether a socket is blocking or not, so we set it to non blocking whatsoever.
-    unsigned long int on = 1;
-    ioctlsocket(s, FIONBIO, &on);
+    if (mIsSocket) {
```

Comment:
I think we should only set the fd to non blocking if it is a socket. Otherwise, it's meaningless. In posix, this has no impact. On windows I don't know.

Reply:
If the file descriptor is not set to non block, read will wait until data is available on Windows. It is not necessary when the caller sets the file descriptor to non block with RsFileUtil::set_fd_nonblock like in TorProcess::start. I have only added the right function for file descriptor, because unix_fcntl_nonblock shows an error in log for a file descriptor.